File: 1347-minimum-number-of-steps-to-make-two-strings-anagram/1347-minimum-number-of-steps-to-make-two-strings-anagram.py
class Solution:
    def minSteps(self, s: str, t: str) -> int:
        freqS = Counter(s)
        ans = 0
        
        for ch in t:
            if freqS[ch]!=0:
                freqS[ch]-=1
            else :
                ans += 1
        for i in freqS.values():
            ans+=i
        
        return ans//2;
            

File: 1347-minimum-number-of-steps-to-make-two-strings-anagram/1347-minimum-number-of-steps-to-make-two-strings-anagram.cpp
class Solution {
public:
    int minSteps(string s, string t) 
    {
        int ans =0;
        unordered_map<char,int> um;
        for(char& ch: s)um[ch]++;
        
        for(char&ch: t)
        {
            if(um[ch]!=0){um[ch]--;if(um[ch]==0)um.erase(ch);}
            else ans++;   
        }
        
        for(auto& it:um)ans+=it.second;
        
        return ans/2;
    }
};